src/components/DateInputs/DateRangeInputs.tsx

```diff
@@ -40,12 +40,27 @@ const DateRangeInputs: React.FC<Props> = (props) => {
 
     const setFromValue = (fromValue: Dayjs | null): void => {
         setLocalFromValue(fromValue);
-        setRangeIfPossible(fromValue, localToValue);
+        if (
+            fromValue &&
+            localToValue &&
+            fromValue > localToValue &&
+            fromValue >= reasonableMinDate
+        ) {
+            setLocalToValue(fromValue);
+            setRangeIfPossible(fromValue, fromValue);
+        } else {
+            setRangeIfPossible(fromValue, localToValue);
+        }
```

Comment:
This method isn't very readable, in what it it is trying to achieve. It feels like a prime candidate for an [Early Return Pattern](https://dev.to/arikaturika/one-concept-a-day-early-return-pattern-in-javascript-3pol)! :)

Let me know if you'd like an example of what that might look like